### ISSUE TITLE
Solar forecast discrepancy: bias correction overcompensating with insufficient data

### ISSUE THREAD

**Issue by jackmcintyre** [jackmcintyre/localshift#756]
## Summary

LocalShift's optimizer is expecting **~1.4-1.5x more solar** than Solcast predicts due to the solar bias correction system overcompensating with insufficient training data.

### Solar Forecast Comparison (Today)

| Time Slot | LocalShift | Solcast | Discrepancy |
|-----------|------------|---------|-------------|
| 10:00-10:30 | 4.03 kWh | 2.79 kWh | +44% |
| 10:30-11:00 | 4.09 kWh | 2.83 kWh | +44% |
| 11:00-11:30 | 4.30 kWh | 2.99 kWh | +44% |

### Solar Accuracy Metrics

| Metric | Value | Interpretation |
|--------|-------|----------------|
| Accuracy | 27.6% | Very poor |
| Bias | -1.76 | Solcast underestimates by 176% |
| MAPE | 72.4% | Mean Absolute Percentage Error |
| Sample count | **4** | ⚠️ Only 4 data points |

## Root Cause Analysis

The solar bias correction system is problematic:

1. **Insufficient training data**: Only 4 samples collected for bias correction
2. **Extreme bias value**: -1.76 suggests Solcast underestimates by 176%
3. **Overcorrection**: The bias correction multiplier is `1.0 - (-1.76) = 2.76x`

**Calculation chain:**
```
Solcast forecast → Code interprets units → Bias correction (2.76x) → Inflated result
```

### Possible Unit Interpretation Issue

The code in `forecast/solar.py` may be incorrectly treating Solcast's `pv_estimate` as kWh/hour when it's actually **kWh per 30-min period**. This would cause the code to divide by 2, then bias correction overcompensates.

## Impact

- Optimizer plans for excess solar that doesn't materialize
- May skip grid charging opportunities expecting solar that won't arrive
- Real-time solar reading (1.255 kW) matches Solcast prediction (1.274 kW), confirming Solcast is accurate

## Recommendations

1. **Disable bias correction** until sufficient samples collected (50+ recommended)
2. **Verify unit interpretation** in `forecast/solar.py` for `pv_estimate`
3. **Add minimum sample threshold** before applying bias correction
4. **Consider resetting solar accuracy tracker** to clear unreliable early data

## Related

- Issue #300 (Express duration inference - recently fixed)
- `custom_components/localshift/forecast/solar.py` - solar bias correction logic
